File: 24_01_week2/wangon/17952.cpp
// 과제는 끝나지 않아!
#include <algorithm>
#include <cstdlib>
#include <iostream>
#include <string>
#include <vector>
using namespace std;

typedef struct subject {
	int score;
	int deadline;
};

vector<subject> arr(1000000);
int ret = 0;

int main()
{
	ios::sync_with_stdio(0);
	cin.tie(0);
	int n;
	cin >> n;
	while (n--)
	{
		int flag;
		cin >> flag;
		if (flag)
		{
			subject obj;
			cin >> obj.score >> obj.deadline;
			arr.push_back(obj);
		}
		arr.back().deadline--;
		if (arr.back().deadline == 0)
		{
			ret += arr.back().score;
			arr.pop_back();
		}
	}
	cout << ret;
}

File: src/AS_24_01_week2/wangon/17952.py
class subject:
    def __init__(self, score, deadline):
        self.score = score
        self.deadline = deadline

n = int(input())
ret = 0
arr = []

for _ in range (n):
    inputs = list(map(int, input().split()))
    if inputs[0]:
        obj = subject(inputs[1], inputs[2])
        arr.append(obj)
    if len(arr) == 0:
        continue
    arr[-1].deadline -= 1
    if arr[-1].deadline == 0:
        ret += arr[-1].score
        arr.pop()
print(ret)
